klayout/drc/run_drc.py

```diff
@@ -0,0 +1,709 @@
+                list_res_db_files.append(future.result())
+            except Exception as exc:
+                logging.error("%s generated an exception: %s" % (run_name, str(exc)))
+                traceback.print_exc()
+
+    ## Check run
+    check_drc_results(list_res_db_files)
+
+
+def run_single_processor(
+    arguments: dict,
+    rule_deck_full_path: str,
+    layout_path: str,
+    switches: dict,
+    drc_run_dir: str,
+):
+    """
+    run_single_processor run the drc checks as single run.
+
+    Parameters
+    ----------
+    arguments : dict
+        Dictionary that holds the arguments passed to the run_drc script.
+    rule_deck_full_path : str
+        String that holds the path of the rule deck files.
+    layout_path : str
+        Path to the target layout.
+    switches : dict
+        Dictionary that holds all the switches that will be passed to klayout run.
+    drc_run_dir : str
+        Path to the run location.
+    """
+
+    list_res_db_files = []
+
+    ## Run Antenna if required.
+    if arguments["--antenna"] or arguments["--antenna_only"]:
+        drc_path = os.path.join(rule_deck_full_path, "rule_decks", "antenna.drc")
+        list_res_db_files.append(
+            run_check(drc_path, "antenna", layout_path, drc_run_dir, switches)
+        )
+
+        if arguments["--antenna_only"]:
+            logging.info("## Completed running Antenna checks only.")
+            exit()
+
+    ## Run Density if required.
+    if arguments["--density"] or arguments["--density_only"]:
+        drc_path = os.path.join(rule_deck_full_path, "rule_decks", "density.drc")
+        list_res_db_files.append(
+            run_check(drc_path, "density", layout_path, drc_run_dir, switches)
+        )
+
+        if arguments["--density_only"]:
+            logging.info("## Completed running density checks only.")
+            exit()
+
+    ## Generate run rule deck from template.
+    if not arguments["--table"]:
+        drc_file = generate_drc_run_template(rule_deck_full_path, drc_run_dir)
+    else:
+        drc_file = generate_drc_run_template(
+            rule_deck_full_path, drc_run_dir, arguments["--table"]
+        )
+
+    ## Run Main DRC
+    list_res_db_files.append(
+        run_check(drc_file, "main", layout_path, drc_run_dir, switches)
+    )
+
+    ## Check run
+    check_drc_results(list_res_db_files)
+
+
+def main(drc_run_dir: str, now_str: str, arguments: dict):
+    """
+    main function to run the DRC.
+
+    Parameters
+    ----------
+    drc_run_dir : str
+        String with absolute path of the full run dir.
+    now_str : str
+        String with the run name for logs.
+    arguments : dict
+        Dictionary that holds the arguments used by user in the run command. This is generated by docopt library.
+    """
+
+    # Check gds file existance
+    if os.path.exists(arguments["--path"]):
+        pass
+    else:
+        logging.error("The input GDS file path doesn't exist, please recheck.")
```

Comment:
would be nice to add the actual path to the error message.

Reply:
done